csrc/executor.cpp

```diff
@@ -257,11 +260,30 @@ void FusionExecutor::compileFusion(
   NVF_ERROR(
       !fusion->outputs().empty(), "No output found for this kernel, aborting.");
 
+  // TODO: refactor the options_ passed through
+  options_.device = c10::Device(c10::DeviceType::CUDA, args.getDeviceIndex());
+
   if (isExpressionEvaluated(fusion)) {
     fusion_ = std::make_unique<Fusion>(*fusion);
     return;
   }
 
+  const auto& exprs = fusion->exprs();
+  if (std::all_of(exprs.begin(), exprs.end(), [](Expr* e) {
+        return isLowerableToCommunication(e);
+      })) {
+    host_ir_container_ = std::make_unique<hir::HostIrContainer>();
+    IrCloner cloner = Fusion::copy(fusion, host_ir_container_.get());
```

Comment:
That was my first attempt before I realized I had to clone the I/O TVs as well. Otherwise, `runFusion` would fail to locate the input tensor via the expression evaluator. 

(I was under the impression that `cloner.clone(e)` clones recursively, but I was wrong)

Reply:
ok!